xpra/platform/xposix/paths.py

```diff
@@ -144,6 +144,9 @@ def do_get_socket_dirs():
     #Debian and Ubuntu often don't create a reliable XDG_RUNTIME_DIR
     #other distros may not create one when using "su"
     SOCKET_DIRS.append("~/.xpra")
+    # Same applies to /run/user/$UID/xpra
+    # (where $UID is expandable environment variable!)
+    SOCKET_DIRS.append("/run/user/$UID/xpra")
```

Comment:
`/run/user/$UID/xpra` is already meant to be added in `do_get_socket_dirs`:
```
    SOCKET_DIRS.append(runtime_dir)`
```
Why isn't this working?
Perhaps `$XDG_RUNTIME_DIR` is not defined?

Reply:
I added some logging at DotXpra:

```diff
--- a/xpra/platform/dotxpra.py	2021-05-25 11:21:25.088428502 +0300
+++ b/platform/dotxpra.py	2021-05-26 09:22:42.283280545 +0300
@@ -45,8 +45,12 @@
                 sockdir = "undefined"
         elif sockdir not in sockdirs:
             sockdirs.insert(0, sockdir)
+        get_util_logger().warn("DotXpra(): sockdir: %s", sockdir)
+        get_util_logger().warn("DotXpra(): sockdirs: %s", sockdirs)
         self._sockdir = self.osexpand(sockdir)
         self._sockdirs = [self.osexpand(x) for x in sockdirs]
+        get_util_logger().warn("DotXpra(): self._sockdir: %s", self._sockdir)
+        get_util_logger().warn("DotXpra(): self._sockdirs: %s", self._sockdirs)
 
     def osexpand(self, v):
         return osexpand(v, self.username, self.uid, self.gid)
```
and reproduced starting the new session using HTML5 client:

```
тра 26 09:23:03 buildbot xpra[92843]: xpra proxy version 4.3 64-bit
тра 26 09:23:03 buildbot xpra[92843]:  uid=0 (root), gid=0 (root)
тра 26 09:23:03 buildbot xpra[92843]:  running with pid 92843 on Linux Debian 11 bullseye
тра 26 09:23:18 buildbot xpra[92843]: Authentication required by PAM authenticator module 1
тра 26 09:23:18 buildbot xpra[92843]:  sending challenge for username 'test' using xor digest
тра 26 09:23:18 buildbot xpra[92843]: DotXpra(): sockdir: $XDG_RUNTIME_DIR/xpra
тра 26 09:23:18 buildbot xpra[92843]: DotXpra(): sockdirs: ['$XDG_RUNTIME_DIR/xpra', '/run/xpra', '~/.xpra']
тра 26 09:23:18 buildbot xpra[92843]: DotXpra(): self._sockdir: $XDG_RUNTIME_DIR/xpra
тра 26 09:23:18 buildbot xpra[92843]: DotXpra(): self._sockdirs: ['$XDG_RUNTIME_DIR/xpra', '/run/xpra', '/home/test/.xpra']
тра 26 09:23:18 buildbot xpra[92843]: DotXpra(): sockdir: $XDG_RUNTIME_DIR/xpra
тра 26 09:23:18 buildbot xpra[92843]: DotXpra(): sockdirs: ['$XDG_RUNTIME_DIR/xpra', '/run/xpra', '~/.xpra']
тра 26 09:23:18 buildbot xpra[92843]: DotXpra(): self._sockdir: $XDG_RUNTIME_DIR/xpra
тра 26 09:23:18 buildbot xpra[92843]: DotXpra(): self._sockdirs: ['$XDG_RUNTIME_DIR/xpra', '/run/xpra', '/home/test/.xpra']
тра 26 09:23:19 buildbot xpra[93019]: Entering daemon mode; any further errors will be reported to:
тра 26 09:23:19 buildbot xpra[93019]:   /run/user/1000/xpra/S93016.log
тра 26 09:23:20 buildbot xpra[93019]: Actual display used: :4
тра 26 09:23:20 buildbot xpra[93019]: Actual log file name is now: /run/user/1000/xpra/:4.log
тра 26 09:23:22 buildbot xpra[92843]: DotXpra(): sockdir: $XDG_RUNTIME_DIR/xpra
тра 26 09:23:22 buildbot xpra[92843]: DotXpra(): sockdirs: ['$XDG_RUNTIME_DIR/xpra', '/run/xpra', '~/.xpra']
тра 26 09:23:22 buildbot xpra[92843]: DotXpra(): self._sockdir: $XDG_RUNTIME_DIR/xpra
тра 26 09:23:22 buildbot xpra[92843]: DotXpra(): self._sockdirs: ['$XDG_RUNTIME_DIR/xpra', '/run/xpra', '/root/.xpra']
тра 26 09:23:22 buildbot xpra[92843]: server socket for display :4 not found
тра 26 09:23:22 buildbot xpra[92843]:  waiting up to 20 seconds
тра 26 09:23:42 buildbot xpra[92843]: Error: cannot start proxy connection:
тра 26 09:23:42 buildbot xpra[92843]:  cannot find live server for display :4
тра 26 09:23:42 buildbot xpra[92843]:  connection definition:
тра 26 09:23:42 buildbot xpra[92843]:  * display            : :4
тра 26 09:23:42 buildbot xpra[92843]:  * display_name       : :4
тра 26 09:23:42 buildbot xpra[92843]:  * gid                : 1000
тра 26 09:23:42 buildbot xpra[92843]:  * local              : True
тра 26 09:23:42 buildbot xpra[92843]:  * socket_dirs        : $XDG_RUNTIME_DIR/xpra, /run/xpra, ~/.xpra
тра 26 09:23:42 buildbot xpra[92843]:  * type               : unix-domain
тра 26 09:23:42 buildbot xpra[92843]:  * uid                : 1000
```
It seems that basic PAM environment block lacks `$XDG_RUNTIME_DIR`.

Running `xpra stop :4` in my host X session expands `$XDG_RUNTIME_DIR` correctly:

```
DotXpra(): sockdir: $XDG_RUNTIME_DIR/xpra
DotXpra(): sockdirs: ['$XDG_RUNTIME_DIR/xpra', '/run/xpra', '~/.xpra']
DotXpra(): self._sockdir: /run/user/1000/xpra
DotXpra(): self._sockdirs: ['/run/user/1000/xpra', '/run/xpra', '/home/test/.xpra']
server requested disconnect:
 server shutdown
DotXpra(): sockdir: /run/user/1000/xpra
DotXpra(): sockdirs: ['/run/user/1000/xpra']
DotXpra(): self._sockdir: /run/user/1000/xpra
DotXpra(): self._sockdirs: ['/run/user/1000/xpra']
xpra at :4 has exited.
```